frontend2/src/views/Login.tsx

```diff
@@ -1,7 +1,105 @@
-import React from "react";
+import React, { useContext, useEffect, useState } from "react";
+import Input from "../components/elements/Input";
+import Button from "../components/elements/Button";
+import { login as apiLogin } from "../utils/api/auth";
+import { type SubmitHandler, useForm } from "react-hook-form";
+import { FIELD_REQUIRED_ERROR_MSG } from "../utils/constants";
+import { Link, useNavigate } from "react-router-dom";
+import { EpisodeContext } from "../contexts/EpisodeContext";
+import { useCurrentUser, AuthStateEnum } from "../contexts/CurrentUserContext";
+import { type Maybe } from "../utils/utilTypes";
+import { getUserUserProfile } from "../utils/api/user";
+
+interface LoginFormInput {
+  username: string;
+  password: string;
+}
 
 const Login: React.FC = () => {
-  return <p>login page</p>;
+  const { register, handleSubmit } = useForm<LoginFormInput>();
+  const { episodeId } = useContext(EpisodeContext);
+  const { login, authState } = useCurrentUser();
+  const [loginError, setLoginError] = useState<Maybe<string>>();
+
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    // redirect to home if already logged in
+    if (authState === AuthStateEnum.AUTHENTICATED) {
+      navigate(episodeId !== undefined ? `/${episodeId}/home` : "/");
+    }
+  }, [authState]);
+
+  const onSubmit: SubmitHandler<LoginFormInput> = async (data) => {
+    try {
+      setLoginError(undefined);
+      await apiLogin(data.username, data.password);
+      login(await getUserUserProfile());
+    } catch (error) {
+      console.log(error);
```

Comment:
Do you think it would be useful to send the specific (i.e. username/pass incorrect) errors to the form elements? Thinking if our jwt errors are a consistent format, then we can use that to determine which is incorrect? If not, not needed :+1:

Reply:
The errors are of a consistent type: `ResponseError` if the server returns a status <200 or >=300, which contains the `response` and a `msg`. 

unfortunately the server doesn't tell us if the username or password is incorrect, so we can't display that info to the user